Idref.js

```diff
@@ -78,12 +82,28 @@ function scrape(doc, url) {
 	var translator = Zotero.loadTranslator('web');	
 	
 	if (url.includes("archives-ouvertes")){
-		// HAL Archives ouvertes
 		translator.setTranslator('58ab2618-4a25-4b9b-83a7-80cd0259f896');
 	} else if (url.includes("sudoc.abes.fr")) {
-		// Sudoc
 		translator.setTranslator('1b9ed730-69c7-40b0-8a06-517a89a3a278');
+	} else if (url.includes("catalogue.bnf.fr")) {
+		translator.setTranslator('47533cd7-ccaa-47a7-81bb-71c45e68a74d');
+	} else if (url.includes("www.theses.fr")) {
+		translator.setTranslator('3f73f0aa-f91c-4192-b0d5-907312876cb9');
 	} else {
+		// Orcid is the only case where we need to use an import translator, different behvior from previous ones
+		if (url.includes("pub.orcid.org")) {
+			// Idrefs returns orcid links with /works/ when the translator only works with the /work/ version of URLs
+			url = url.replace("/works/", "/work/");
+
+			ZU.doGet(url, function(text){
+				var translator = Zotero.loadTranslator("import");
+				translator.setTranslator("bc03b4fe-436d-4a1f-ba59-de4d2d7a63f7");//CSL JSON
+				translator.setString(text);
+				translator.translate();
+			}, undefined, undefined, {"Accept" : "application/vnd.citationstyles.csl+json"});
+	
+			return true;
```

Comment:
Use simply `return;` for stopping here. (This holds also for the other one below.)